src/colmap/math/math.h

```diff
@@ -166,7 +167,7 @@ bool NextCombination(Iterator first1,
 
 template <typename T>
 int SignOfNumber(const T val) {
-  return (T(0) < val) - (val < T(0));
+  return val >= 0 ? 1 : -1;
```

Comment:
Only used in the homography decomposition. There are some rare edge cases, where the previous implementation would return 0 (~2-3% of cases) and then the decomposition falsely returns identical rotations for all 4 solutions.